### ISSUE TITLE
searchResults methods on a Paginated result

### ISSUE THREAD

**Issue by Surt** [babenkoivan/elastic-scout-driver-plus#43]
Hi!

me again. I'm searching using paginated results and using the new and marvelous `Match::raw` method. Everything is fine.
The problem is: I can't eagerload or lazy load nothing inside the paginated results.

```php
$paginator = MyModel::boolSearch()
    ->must('match_all')
    ->paginate(10);
```

so, the paginator collection is not a `searchResult` collection and does not have the `$paginator ->getCollection()->models()' 
My only option right now is to transverse each match and lazy load on each model wich is not enought.

If I find some time I wil try to set the paginated collection = to a searchresult and send you reports.

**Comment by Surt** [comment]
ok, so I did a quick changes just to try:

first, I created 
```php
<?php declare(strict_types=1);

namespace ElasticScoutDriverPlus\Support;

use ElasticScoutDriverPlus\Match;
use Illuminate\Database\Eloquent\Collection as EloquentCollection;
use Illuminate\Support\Collection;

class MatchsCollection extends Collection
{
    public function models(): EloquentCollection
    {
        $models = new EloquentCollection();

        $this->each(static function (Match $match) use ($models) {
            $models->push($match->model());
        });

        return $models->filter()->values();
    }
}

```

and on `ElasticScoutDriverPlus\SearchResult` I changed (just to try) the `matches` method to return this custom collection

```php 

    public function matches(): MatchsCollection
    {
        return new MatchsCollection($this->matches);
    }
``

Then, on the `paginate` method (ElasticScoutDriverPlus\Builders\SearchRequestBuilder`  I send a collection instead of array (this way the `LenghtAwarePaginator` uses that collection:
```php
        return new LengthAwarePaginator(
            $searchResult->matches(), // matches become the custom collection with the models method
            $searchResult->total(),
            $perPage,
            $page,
            [
                'path' => Paginator::resolveCurrentPath(),
                'pageName' => $pageName,
            ]
        );
```


This is just a quick test to see if it is possible. Thankfully your code @babenkoivan is as good as always!

What do you think? Should I try to send you a merge request with the formal solution?

**Comment by babenkoivan** [comment]
Hey @Surt, I have plans to improve pagination. Most likely, `SearchResult` and a new `Paginator` will be implementing the same interface and therefore have the same set of methods: `models()`, `documents()`, etc.

But you can already get models from a paginator without any code changes: `collect($paginator->items())->map->model();`

**Comment by babenkoivan** [comment]
Hey @Surt, I've just released [v1.17.0](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v1.17.0). You can now use `SearchResult` methods on paginated results:

```php
$paginator = MyModel::boolSearch()
    ->must('match_all')
    ->paginate(10);

$models = $paginator->models();
```

**Comment by Surt** [comment]
Looking forward to try it (and nose around the code :p) Thanks @babenkoivan
